### PR TITLE
refactor: rename bitDepth to depth

### LINKED ISSUE #444: Exporting 16-bits images as png doesn't work
In the function `encodePng` that uses `fast-png` library : 
```
encode(image.getRawImage(), options);
```
We have :
```
Image.getRawImage(): {
    width: number;
    height: number;
    data: ImageDataArray;
    channels: number;
    bitDepth: BitDepth;
}
```

While : 
```
export interface ImageData {
    width: number;
    height: number;
    data: PngDataArray;
    depth?: BitDepth;
    channels?: number;
}
```

`bitDepth` in `getRawImage` has to be changed to `depth`
So here since we don't pass a value to `depth` it uses the default value in `fast-png` which is 8 bits

### PR DESCRIPTION
**Pull request by moonayyur**
closes #444

### CODE REVIEW

**Comment by stropitek** [comment]
Would it be simpler to remap `bitDepth` to `depth` when passing the options to fast-png?

**Review by targos — CHANGES_REQUESTED** [review]
I'm sorry but I don't see a good reason for making this big change.

**Comment by stropitek** [comment]
> Would it be simpler to remap `bitDepth` to `depth` when passing the options to fast-png?

Simpler but mainly non-breaking

File: src/save/encodePng.ts [hunk 1]
```diff
@@ -14,5 +14,14 @@ export function encodePng(
...
 ): Uint8Array {
-  return encode(image.getRawImage(), options);
+  return encode(
+    {
+      data: image.getRawImage().data,
+      width: image.getRawImage().width,
+      height: image.getRawImage().height,
+      channels: image.getRawImage().channels,
+      depth: image.getRawImage().bitDepth,
+    },
```
Comment (targos) [inline src/save/encodePng.ts:24]: Can you avoid the repetition with a variable?

**Comment by stropitek** [comment]
You can merge with main to get rid of the CI errors

**Comment by targos** [comment]
Can you please add a test that would fail before the fix?

**Review by targos — CHANGES_REQUESTED** [review]
You fix the `encodePng` function, so the test should execute that function, not the one from `fast-png` directly.

**Comment by moonayyur** [comment]
> You fix the `encodePng` function, so the test should execute that function, not the one from `fast-png` directly.

So the image must be converted to 8 bits first ? Since the `encodePng` function only accepts a prop of type `Image` where `bitDepth` is mandatory (so we can't test directly with the default value of `fast-png`)

**Comment by targos** [comment]
I don't see why it would have to be converted to 8 bits first. `bitDepth` is mandatory, but its value doesn't have to be 8.

**Comment by moonayyur** [comment]
Oh, okay I had misunderstood this 
> Can you please add a test that would fail before the fix?

I thought I should write a test that shows the previous error

**Comment by targos** [comment]
You should write a test that pass with the fix and doesn't pass without the fix.

**Comment by moonayyur** [comment]
The previous error was that the attribute `depth` in `fast-png` wasn't getting any value from `img` since it was named `bitDepth`
But now `bitDepth` is renamed to `depth` inside `encodePng`, so I don't see how I could go back to the error in the test

**Comment by targos** [comment]
You're correct, and that's not what I'm asking.
Somehow I'm not making myself clear but I don't know how to say it differently 😕

**Review by targos — APPROVED** [review]

**Pull request merged**